chains/solana/contracts/programs/ccip-router/src/lib.rs

```diff
@@ -669,6 +669,51 @@ pub mod ccip_router {
         .amount)
     }
 
+    /// Transfers the accumulated billed fees in a particular token to an arbitrary token account.
+    /// Only the CCIP Admin can withdraw billed funds.
+    ///
+    /// # Arguments
+    ///
+    /// * `ctx` - The context containing the accounts required for the transfer of billed fees.
+    /// * `transfer_all` - A flag indicating whether to transfer all the accumulated fees in that token or not.
+    /// * `desired_amount` - The amount to transfer. If `transfer_all` is true, this value must be 0.
+    pub fn withdraw_billed_funds(
+        ctx: Context<WithdrawBilledFunds>,
+        transfer_all: bool,
+        desired_amount: u64, // send 0 if transfer_all is false
```

Comment:
This comment is a bit ambiguous. It's not clear whether the caller is being asked to pass 0 if `transfer_all` is false, or whether the execution will cause 0 funds to be sent. The wording on the docstring "If `transfer_all` is true, this value must be 0" is much clearer.